src/webgpu/idl/javascript.spec.ts

```diff
@@ -0,0 +1,339 @@
+        desc = descriptorModifier?.descriptorModifier
+          ? descriptorModifier.descriptorModifier(adapter, desc)
+          : desc;
+        return addAllFeatures(adapter, desc);
+      },
+      keyModifier(baseKey: string) {
+        return `${baseKey}:AllFeaturesTest`;
+      },
+    };
+    super.requestDeviceWithRequiredParametersOrSkip(
+      initUncanonicalizedDeviceDescriptor(descriptor),
+      mod
+    );
+  }
+}
+
+/**
+ * A Test that requests all the max limits from the adapter on the device.
+ */
+class AllFeaturesTest extends GPUTest {
+  public static override MakeSharedState(
+    recorder: TestCaseRecorder,
+    params: TestParams
+  ): GPUTestSubcaseBatchState {
+    return new AllFeaturesGPUTestSubcaseBatchState(recorder, params);
+  }
+}
+
+const kResourceInfo = {
+  buffer: {
+    create(t: GPUTest) {
+      return t.createBufferTracked({ size: 16, usage: GPUBufferUsage.UNIFORM });
+    },
+    requiredKeys: [
+      'destroy',
+      'getMappedRange',
+      'label',
+      'mapAsync',
+      'mapState',
+      'size',
+      'unmap',
+      'usage',
+    ],
+  },
+  texture: {
+    create(t: GPUTest) {
+      return t.createTextureTracked({
+        size: [2, 3],
+        format: 'r8unorm',
+        usage: GPUTextureUsage.TEXTURE_BINDING,
+      });
+    },
+    requiredKeys: [
+      'createView',
+      'depthOrArrayLayers',
+      'destroy',
+      'dimension',
+      'format',
+      'height',
+      'label',
+      'mipLevelCount',
+      'sampleCount',
+      'usage',
+      'width',
+    ],
+  },
+  querySet: {
+    create(t: GPUTest) {
+      return t.createQuerySetTracked({
+        type: 'occlusion',
+        count: 2,
+      });
+    },
+    requiredKeys: ['count', 'destroy', 'label', 'type'],
+  },
+  adapter: {
+    create(t: GPUTest) {
+      return t.adapter;
+    },
+    requiredKeys: [
+      'featureLevel',
+      'features',
+      'info',
+      'isCompatibilityMode',
+      'isFallbackAdapter',
+      'limits',
+      'requestDevice',
+    ],
+  },
+  device: {
+    create(t: GPUTest) {
+      return t.device;
+    },
+    requiredKeys: [
+      'adapterInfo',
+      'addEventListener',
+      'createBindGroup',
+      'createBindGroupLayout',
+      'createBuffer',
+      'createCommandEncoder',
+      'createComputePipeline',
+      'createComputePipelineAsync',
+      'createPipelineLayout',
+      'createQuerySet',
+      'createRenderBundleEncoder',
+      'createRenderPipeline',
+      'createRenderPipelineAsync',
+      'createSampler',
+      'createShaderModule',
+      'createTexture',
+      'destroy',
+      'dispatchEvent',
+      'features',
+      'importExternalTexture',
+      'label',
+      'limits',
+      'lost',
+      'onuncapturederror',
+      'popErrorScope',
+      'pushErrorScope',
+      'queue',
+      'removeEventListener',
+    ],
+  },
+  'adapter.limits': {
+    create(t: GPUTest) {
+      return t.adapter.limits;
+    },
+    requiredKeys: kLimits,
```

Comment:
Same for `*In{Vertex,Fragment}Stage` limits

Reply:
fixed